imap/xapian_wrap.cpp

```diff
@@ -700,7 +699,9 @@ EXPORTED int xapian_dbw_open(const char **paths, xapian_dbw_t **dbwp,
     std::set<int> db_versions;
     try {
         int flags = Xapian::DB_BACKEND_GLASS|Xapian::DB_RETRY_LOCK;
-        if (nosync) flags |= Xapian::DB_DANGEROUS|Xapian::DB_NO_SYNC;
+        if (mode & XAPIAN_DBW_NOSYNC) {
+            flags |= Xapian::DB_DANGEROUS|Xapian::DB_NO_SYNC;
```

Comment:
Do we also need something like `mode &= ~XAPIAN_DBW_NOSYNC;` in here?  I notice later on around line 740 we do a `mode == ...` comparison, and that would now fail if this flag has been or'd in.